cosmos/testing/integration/precompile/misc/misc_test.go

```diff
@@ -21,14 +21,73 @@
 package misc_test
 
 import (
+	"math/big"
+	"reflect"
 	"testing"
 
 	"pkg.berachain.dev/polaris/cosmos/testing/integration"
+	"pkg.berachain.dev/polaris/eth/accounts/abi"
+	"pkg.berachain.dev/polaris/eth/common/hexutil"
 
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
 )
 
+type MsgSendEnergy struct {
+	To     *big.Int `json:"to"`
+	From   *big.Int `json:"from"`
+	Amount *big.Int `json:"amount"`
+}
+
+func TestDecoding(t *testing.T) {
+	RegisterTestingT(t)
+	// bytes from abi.encoding the following solidity struct:
+	//    struct MsgSendEnergy {
+	//        uint to;
+	//        uint from;
+	//        uint amount;
+	//    }
+	// with values: 3293, 2385, 23509
+	expectedMsg := MsgSendEnergy{
+		To:     big.NewInt(3293),
+		From:   big.NewInt(2385),
+		Amount: big.NewInt(23509),
+	}
+	bzStr := "0x0000000000000000000000000000000000000000000000000000000000000cdd00000000000000000000000000000000000000000000000000000000000009510000000000000000000000000000000000000000000000000000000000005bd5"
+	bz, err := hexutil.Decode(bzStr)
+	Expect(err).To(BeNil())
+
+	// make MsgSendEnergy abi type.
+	msgSendEnergyType, err := abi.NewType("tuple", "", []abi.ArgumentMarshaling{
```

Comment:
@technicallyty do you know if this is possible with the structs generated in `abigen`

Reply:
we'd currently have to recreate these everytime, though I guess you could rwite a generator

Reply:
can import this type directly from the solidity contract's go bindings, so not needed manually here

Reply:
@calbera are you referring to the go bindings ala `i_governance_module.abigen.go`? if so, that doesn't work as it's just a Go struct, and abi package doesn't know how to unmarshal abi encoded bytes into Go structs, only abi types. 

unless theres somewhere in abigen thats making these abi types that im not seeing?

Reply:
so if you have the .abi.json file of your solidity contract/interface, you can build the abi.ABI struct which has all of this generated for you. Get the Methods --> Inputs/Outputs --> Type

Reply:
for the go struct, we have to do the same thing you have here https://github.com/berachain/polaris/blob/main/cosmos/lib/conversions.go#L54-L59